optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/move/Move.java

```diff
@@ -87,6 +87,18 @@
      */
     Move<Solution_> doMove(ScoreDirector<Solution_> scoreDirector);
 
+    /**
+     * As defined by {@link #doMove(ScoreDirector)}, but does not return an undo move.
+     * It is recommended that {@link #doMove(ScoreDirector)} is implemented by wrapping {@link #doMoveOnly(ScoreDirector)}.
+     *
+     * @param scoreDirector never null, the {@link ScoreDirector} that needs to get notified of the changes
+     */
+    default void doMoveOnly(ScoreDirector<Solution_> scoreDirector) {
+        // For backwards compatibility, this method is default and calls doMove(...).
+        // In reality, the relationship would be inversed, as implemented in AbstractMove.
```

Comment:
(nitpick) pressgang/ibm/etc docs style guidelines: avoid "would", etc as they are ambiguous signals

```suggestion
        // Normally, the relationship is typically inversed, as implemented in AbstractMove.
```